plugins/spi/SPIOutput.h

```diff
@@ -214,6 +217,91 @@ class SPIOutput: public ola::rdm::DiscoverableRDMControllerInterface {
   static const uint16_t APA102_SPI_BYTES_PER_PIXEL;
   static const uint16_t APA102_START_FRAME_BYTES;
   static const uint8_t APA102_LEDFRAME_START_MARK;
+  static const uint16_t TLC5971_SLOTS_PER_DEVICE;
+  static const uint16_t TLC5971_SPI_BYTES_PER_DEVICE;
+
+  // TLC5971 data structure
+  PACK(
+  struct TLC5971_packet_config_fields_t{
+    //  Write Command (6Bit)
+    uint8_t WRCMD : 6;
+    //  Function Control Data (5 x 1Bit = 5Bit)
+    uint8_t OUTTMG : 1;
+    uint8_t EXTGCK : 1;
+    uint8_t TMGRST : 1;
+    uint8_t DSPRPT : 1;
+    uint8_t BLANK : 1;
+    //  BC-Data (3 x 7Bits = 21Bit)
+    uint8_t BCB : 7;
+    uint8_t BCG : 7;
+    uint8_t BCR : 7;
+  });
+
+  union TLC5971_packet_config_t {
+    uint8_t bytes[4];
+    // 6 + 5 + 21 = 4byte
+     TLC5971_packet_config_fields_t fields;
+  };
+
+  // //  Write Command (6Bit)
+  // static const uint8_t TLC5971_PACKET_CONFIG_MASKS_WRCMD = 0b00111111;
+  // //  Function Control Data (5 x 1Bit = 5Bit)
+  // static const uint8_t TLC5971_PACKET_CONFIG_MASKS_OUTTMG = 0b00000001;
+  // static const uint8_t TLC5971_PACKET_CONFIG_MASKS_EXTGCK = 0b00000001;
+  // static const uint8_t TLC5971_PACKET_CONFIG_MASKS_TMGRST = 0b00000001;
+  // static const uint8_t TLC5971_PACKET_CONFIG_MASKS_DSPRPT = 0b00000001;
+  // static const uint8_t TLC5971_PACKET_CONFIG_MASKS_BLANK  = 0b00000001;
+  // //  BC-Data (3 x 7Bits = 21Bit)
+  // static const uint8_t TLC5971_PACKET_CONFIG_MASKS_BCB = 0b01111111;
+  // static const uint8_t TLC5971_PACKET_CONFIG_MASKS_BCG = 0b01111111;
+  // static const uint8_t TLC5971_PACKET_CONFIG_MASKS_BCR = 0b01111111;
+  //
+  // //  Write Command (6Bit)
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_WRCMD  = 2;
+  // //  Function Control Data (5 x 1Bit = 5Bit)
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_OUTTMG = 1;
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_EXTGCK = 0;
+  // // byte border ------------------------------------------
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_TMGRST = 7;
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_DSPRPT = 6;
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_BLANK  = 5;
+  // //  BC-Data (3 x 7Bits = 21Bit)
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_BCB_RS = 2;
+  // // byte border ------------------------------------------
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_BCB_LS = 6;
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_BCG_RS = 1;
+  // // byte border ------------------------------------------
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_BCG_LS = 7;
+  // static const uint8_t TLC5971_PACKET_CONFIG_LSHIFT_BCR = 0;
+
+  union TLC5971_packet_gsdata_t {
+    uint8_t bytes[24];
+    // the uint16_t will not work everywhere because of endianess problems..
```

Comment:
SPaG: endianness

Reply:
i will have a look at this..

Reply:
http://docs.openlighting.org/ola/doc/latest/namespaceola_1_1network.html#ae9263a09db6563136f1e46af389dbf0f